File: src/main/java/com/github/zametki/ajax/BaseGroupActionAjaxCall.java
package com.github.zametki.ajax;

import com.github.zametki.Context;
import com.github.zametki.UserSession;
import com.github.zametki.model.Group;
import com.github.zametki.model.GroupId;
import com.github.zametki.model.UserId;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public abstract class BaseGroupActionAjaxCall extends BaseAjaxCall {

    protected UserId userId;

    @NotNull
    @Override
    protected String getResponseText() throws Exception {
        userId = UserSession.get().getUserId();
        if (userId == null) {
            return permissionError();
        }
        Group group = Context.getGroupsDbi().getById(new GroupId(getParameter("groupId").toInt(-1)));
        if (group != null && !group.userId.equals(userId)) {
            return permissionError();
        }
        return getResponseText(group);
    }

    protected abstract String getResponseText(@Nullable Group group) throws Exception;
}

File: src/main/java/com/github/zametki/ajax/BaseGroupActionAjaxCall.kt
package com.github.zametki.ajax

import com.github.zametki.Context
import com.github.zametki.UserSession
import com.github.zametki.model.Group
import com.github.zametki.model.GroupId
import com.github.zametki.model.UserId

abstract class BaseGroupActionAjaxCall : BaseAjaxCall() {

    protected var userId: UserId = UserId.INVALID_ID

    @Throws(Exception::class)
    override fun getResponseText(): String {
        userId = UserSession.get().userId ?: UserId.INVALID_ID
        if (userId.isInvalid) {
            return permissionError()
        }
        val group = Context.getGroupsDbi().getById(GroupId(getParameter("groupId").toInt(-1)))
        return if (group != null && group.userId != userId) {
            permissionError()
        } else getResponseText(group)
    }

    @Throws(Exception::class)
    protected abstract fun getResponseText(group: Group?): String
}
